python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -174,6 +174,24 @@ def get_full_subset(self, *, key: T_EntityKey) -> EntitySubset[T_EntityKey]:
         )
         return EntitySubset(self, key=key, value=_ValidatedEntitySubsetValue(value))
 
+    @use_partition_loading_context
+    def get_subset_not_in_graph(
+        self, *, key: T_EntityKey, candidate_subset: EntitySubset[T_EntityKey]
+    ) -> EntitySubset[T_EntityKey]:
+        partitions_def = self._get_partitions_def(key)
+        check.invariant(
+            partitions_def is not None and candidate_subset.is_partitioned,
```

Comment:
which means this check is potentially a bit redundant, but certainly shouldn't hurt